### PR TITLE
Kiran/fix/s3encryption removal

### PR DESCRIPTION
**Pull request by kiranthumma**
## Description of this change
remove s3 kms encryption

## Why is this change being made?
- [x] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## How was this tested? How can the reviewer verify your testing?
terraform plan in both dev and prod

## Related issues
n/a

## Checklist
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.
- [x] I have evaluated the security impact of this change, and [OWASP Secure Coding Practices](https://owasp.org/www-project-secure-coding-practices-quick-reference-guide/migrated_content#) have been observed.
- [x] I have informed stakeholders of my changes.

### CODE REVIEW

**Comment by hanpeter** [comment]
Please run `terraform fmt`

File: main.tf [hunk 1]
```diff
@@ -227,11 +215,16 @@ data "aws_iam_policy_document" "bastion_host_policy_document" {
...
+      "s3:PutObject"
     ]
-    resources = [aws_kms_key.key.arn]
+    resources = ["${aws_s3_bucket.bucket.arn}/*"]
+
+    condition {
+      test     = "StringEquals"
+      variable = "s3:x-amz-server-side-encryption"
+      values   = ["AES256"]
+    }
```
Comment (hanpeter) [inline main.tf:229]: Can you check whether this condition is needed? We have a bunch of S3 bucket creations in our main Terraform repository.

File: main.tf [hunk 2]
```diff
@@ -227,11 +215,16 @@ data "aws_iam_policy_document" "bastion_host_policy_document" {
 
   statement {
     actions = [
-
-      "kms:Encrypt",
-      "kms:Decrypt"
+      "s3:GetObject",
+      "s3:PutObject"
```
Comment (hanpeter) [inline main.tf]: Don't you also set ACL when you add the object?

File: main.tf (hunk 1 above)
Comment (kiranthumma) [reply main.tf:229]: For this condition it depends on which  `server_side_encryption_configuration` are we using. If we are using a `sse_algorithm = "AES256"` then this is the condition.

Comment (hanpeter) [reply main.tf:229]: Ah, never mind. I was comparing this to bucket policies. That's probably why we have a difference here. Proceed.

**Review by hanpeter — APPROVED** [review]

**Pull request merged**